examples/apollo-engine/index.js

```diff
@@ -1,33 +1,87 @@
-const { GraphQLServer } = require('graphql-yoga')
-const { Engine } = require('apollo-engine')
-const compression = require('compression')
+const { GraphQLServer, PubSub } = require("graphql-yoga");
+const { ApolloEngine } = require("apollo-engine");
+const compression = require("compression");
 
 const typeDefs = `
   type Query {
     hello(name: String): String!
-  }`
+  }
+
+  type Counter {
+    count: Int!
+    countStr: String
+  }
+
+  type Subscription {
+    counter: Counter!
+  }
+`;
 
 const resolvers = {
   Query: {
-    hello: (_, { name }) => `Hello ${name || 'World'}`,
+    hello: (_, { name }) => `Hello ${name || "World"}`
+  },
+  Counter: {
+    countStr: counter => `Current count: ${counter.count}`
   },
-}
-
-const server = new GraphQLServer({ typeDefs, resolvers })
-
-const engine = new Engine({
-  engineConfig: { apiKey: process.env.APOLLO_ENGINE_KEY },
-  endpoint: '/',
-  graphqlPort: parseInt(process.env.Port, 10) || 4000,
-})
-engine.start()
-
-// Enable gzip compression
-// ref: https://www.apollographql.com/docs/engine/setup-node.html#enabling-compression
-server.express.use(compression())
-server.express.use(engine.expressMiddleware())
-
-server.start({
-  tracing: true,
-  cacheControl: true
-}, () => console.log('Server is running on localhost:4000'))
+  Subscription: {
+    counter: {
+      subscribe: (parent, args, { pubsub }) => {
+        const channel = Math.random().toString(36).substring(2, 15); // random channel name
+        let count = 0;
+        setInterval(
+          () => pubsub.publish(channel, { counter: { count: count++ } }),
+          2000
+        );
+        return pubsub.asyncIterator(channel);
+      }
+    }
+  }
+};
+
+const start = async () => {
+  const pubsub = new PubSub();
+  const port = parseInt(process.env.PORT, 10) || 4000;
+  const server = new GraphQLServer({
+    typeDefs,
+    resolvers,
+    context: { pubsub }
+  });
+
+  const engine = new ApolloEngine({
+    apiKey: process.env.APOLLO_ENGINE_KEY
+  });
+
+  // Enable gzip compression
+  // ref: https://www.apollographql.com/docs/engine/setup-node.html#enabling-compression
+  // server.express.use(compression())
+
+  const withEngine = true;
```

Comment:
Here, you can play activate/disable engine easily to confirm the subscriptions bug 😉 (I will probably keep this but with `process.env.NODE_ENV === "production"` if you are agree)